core/engine.go

```diff
@@ -288,6 +288,14 @@ func (e *Engine) processMetrics(globalCtx context.Context, processMetricsAfterRu
 			if !e.NoThresholds {
 				e.processThresholds()
 			}
+			if iim, ok := e.Metrics[metrics.InterruptedIterations.Name]; ok {
```

Comment:
This map lookup is racy according to the current `TestSentReceivedMetrics` failure. So I guess we'll need to rethink or abandon this hack...

Reply:
I am pretty sure this breaks any output as you are only emitting the `1` to them ...

Reply:
I guess it depends on how this metric is agreggated. If their system allows it they could factor it in with the `iterations` metric and get a percentage that way, which is kind of what we're doing here for the summary.

It makes no sense to me to emit the `0` value of this metric considering it's equivalent to `iterations`, and it would add a considerable amount of output data and affect the performance. A 5s/5VU test with an empty `default` and no sleep does 2115298 iterations on my machine and generates a 1.5 GB JSON file when emitting the `0` value. The same test without it manages to do 3203002 iterations and generate a 1.9 GB JSON file, but this is because it's doing ~50% more iterations.

Considering we already have complaints about outputting too much data and interest in #1321, I think we should rather work on #1321 before we consider adding another default metric.

Though we should probably rethink duplicating `iterations`...

How about instead of a new metric, we simply reused `iterations` and added an `interrupted: "<cause>"` tag to it? If the tag is omitted it's a complete iteration and if it's set then it was interrupted with the specific cause. WYT?

Reply:
I agree with prioritizing #1321, but that will actually probably require quite the refactoring, IMO even for an MVP. 
The reusage of the iterations was discussed (I think) and we decided against it based on the fact the primary usage for this will be to check that a percentage of the iterations aren't actually interrupted, which with the current thresholds isn't possible if the metric isn't rate... so :(. 

 I think that maybe we should just make another PR with the changes that are in this PR (the test fixes and the lib.Exception) and merge that to both have them and make this PR easier to rebase and then work on #1321 so we can merge this then.

Reply:
> The reusage of the iterations was discussed (I think) and we decided against it based on the fact the primary usage for this will be to check that a percentage of the iterations aren't actually interrupted, which with the current thresholds isn't possible if the metric isn't rate

Then maybe that should be a feature? I'll see what I can do.